vm/sdk/src/call.rs

```diff
@@ -12,5 +20,25 @@ pub fn call(address: Address, input: Option<Vec<u8>>, method: Option<&str>, amou
   let input_payload = payload.encode();
   let input_len = input_payload.len();
   let input = crate::bytes_to_u32_vec(input_payload);
-  athena_vm::syscalls::call(address.as_ptr(), input.as_ptr(), input_len, amount.as_ptr());
+
+  // Allocate 1024 bytes on the heap, using u32 for proper alignment
+  let mut output_buf = Vec::<std::mem::MaybeUninit<u32>>::with_capacity(1024 / 4);
+
+  let output_size = athena_vm::syscalls::call(
+    address.as_ptr(),
+    input.as_ptr(),
+    input_len,
+    output_buf.as_mut_ptr() as *mut u32,
+    1024,
+    amount.as_ptr(),
+  );
+
+  // Convert the Vec<MaybeUninit<u32>> into a Vec<u8> without freeing buffer
+  let raw_ptr = output_buf.as_mut_ptr() as *mut u8;
+  let capacity = output_buf.capacity();
+  // Prevent `buffer` from being dropped, as we're transferring ownership
+  std::mem::forget(output_buf);
+
+  // Convert the buffer into Vec<u8>, accounting for the number of bytes written
+  unsafe { Vec::from_raw_parts(raw_ptr, output_size, capacity) }
```

Comment:
It couldn't hurt to add an assert here that `output_size <= capacity`. Your host call implementation doesn't allow this, but it's nice to be explicit when using unsafe code to prevent confusing memory errors down the line if anything changes